### PR TITLE
Added Ubuntu/Gazebo combination check to throw an early warning

### LINKED ISSUE #43: Raise early warnings when using incompatible Ubuntu + Gazebo combinations
## Summary

The current implementation allows users to specify an incompatible Ubuntu + Gazebo combination (eg. Focal + Harmonic) which will cause the action to fail at a later stage. Instead, raise an early warning and and end the job.

## Implementation suggestions

Edit the array holding [valid Gazebo distributions](https://github.com/gazebo-tooling/setup-gazebo/blob/6158ecb505e37f0798940270eacf857296424b42/src/utils.ts#L52-L58) to also store corresponding compatible ubuntu distributions. Check against this object when validating the input Gazebo distribution.

### PR DESCRIPTION
**Pull request by sauk2**
Closes #43 

## Summary

- Added an object that holds the Ubuntu/Gazebo combinations
- Added a distro check for Linux workflow against this internal object
- Updated tests

### CODE REVIEW

**Comment by j-rivero** [comment]
I would like not to code that enum that matches exactly the strings if possible. How about a patch something like:
```diff
diff --git a/src/utils.ts b/src/utils.ts
index 1d2572f..3c55497 100644
--- a/src/utils.ts
+++ b/src/utils.ts
@@ -2,49 +2,32 @@ import * as actions_exec from "@actions/exec";
 import * as core from "@actions/core";
 import * as im from "@actions/exec/lib/interfaces";
 
-// Enum of Gazebo distributions
-enum gazeboDistro {
-       CITADEL = "citadel",
-       FORTRESS = "fortress",
-       GARDEN = "garden",
-       HARMONIC = "harmonic",
-       IONIC = "ionic",
-}
-
-// Enum of Ubuntu distributions
-enum ubuntuDistro {
-       FOCAL = "focal",
-       JAMMY = "jammy",
-       NOBLE = "noble",
-}
-
-// List of Valid Gazebo distributions with compatible
-// Ubuntu distributions
 const validGazeboDistroList: {
-       name: gazeboDistro;
-       compatibleUbuntuDistros: ubuntuDistro[];
-}[] = [
-       {
-               name: gazeboDistro.CITADEL,
-               compatibleUbuntuDistros: [ubuntuDistro.FOCAL],
-       },
-       {
-               name: gazeboDistro.FORTRESS,
-               compatibleUbuntuDistros: [ubuntuDistro.FOCAL, ubuntuDistro.JAMMY],
-       },
-       {
-               name: gazeboDistro.GARDEN,
-               compatibleUbuntuDistros: [ubuntuDistro.FOCAL, ubuntuDistro.JAMMY],
-       },
-       {
-               name: gazeboDistro.HARMONIC,
-               compatibleUbuntuDistros: [ubuntuDistro.JAMMY, ubuntuDistro.NOBLE],
-       },
-       {
-               name: gazeboDistro.IONIC,
-               compatibleUbuntuDistros: [ubuntuDistro.NOBLE],
-       },
-];
+  name: string;
+  compatibleUbuntuDistros: string[];
+}[] =
+  [
+    {
+      name: "citadel",
+      compatibleUbuntuDistros: ["focal"],
+    },
+    {
+      name: "fortress",
+      compatibleUbuntuDistros: ["focal", "jammy"],
+    },
+    {
+      name: "garden",
+      compatibleUbuntuDistros: ["focal", "jammy"],
+    },
+    {
+      name: "harmonic",
+      compatibleUbuntuDistros: ["jammy", "noble"],
+    },
+    {
+      name: "ionic",
+      compatibleUbuntuDistros: ["noble"],
+    },
+  ];
 
 /**
  * Execute a command and wrap the output in a log group.
@@ -143,20 +126,18 @@ export function getRequiredGazeboDistributions(): string[] {
  * @param ubuntuCodename
  */
 export function checkUbuntuCompatibility(
-       requiredGazeboDistributionsList: string[],
-       ubuntuCodename: string,
+    requiredGazeboDistributionsList: string[],
+    ubuntuCodename: string,
 ) {
-       requiredGazeboDistributionsList.forEach((element) => {
-               const idx = validGazeboDistroList.findIndex((obj) => {
-                       return obj.name === element;
-               });
-               const compatibleUbuntuList: string[] = Object.values(
-                       validGazeboDistroList[idx].compatibleUbuntuDistros,
-               );
-               if (compatibleUbuntuList.indexOf(ubuntuCodename) <= -1) {
-                       throw new Error("Incompatible Gazebo and Ubuntu combination.");
-               }
-       });
+    requiredGazeboDistributionsList.forEach((element) => {
+      const idx = validGazeboDistroList.findIndex((obj) => {
+        return obj.name === element;
+      });
+      const compatibleUbuntuList: string[] = validGazeboDistroList[idx].compatibleUbuntuDistros;
+    if (compatibleUbuntuList.indexOf(ubuntuCodename) <= -1) {
+      throw new Error("Incompatible Gazebo and Ubuntu combination.");
+    }
+  });
 }
 
 /**
```
that uses directly the strings?

File: __test__/main.test.ts [hunk 1]
```diff
@@ -95,6 +98,23 @@ describe("validate distribution test", () => {
 	});
 });
 
+describe("workflow test with incompatible Ubuntu combination", () => {
```
Comment (j-rivero) [inline __test__/main.test.ts:101]: test is really cool!

File: src/utils.ts [hunk 2]
```diff
@@ -2,6 +2,50 @@ import * as actions_exec from "@actions/exec";
 import * as core from "@actions/core";
 import * as im from "@actions/exec/lib/interfaces";
 
+// Enum of Gazebo distributions
+enum gazeboDistro {
```
Comment (j-rivero) [inline src/utils.ts:6]: The really awesome thing here would to use https://raw.githubusercontent.com/gazebo-tooling/release-tools/master/jenkins-scripts/dsl/gz-collections.yaml to parse the collections from the YAML and the grab the ubuntu distributions from ci_configs or packaging_configs. We can leave it for a new PR or create an issue by now.

File: src/utils.ts [hunk 3]
```diff
@@ -98,6 +134,31 @@ export function getRequiredGazeboDistributions(): string[] {
...
+) {
+	requiredGazeboDistributionsList.forEach((element) => {
+		const idx = validGazeboDistroList.findIndex((obj) => {
+			return obj.name === element;
+		});
+		const compatibleUbuntuList: string[] = Object.values(
+			validGazeboDistroList[idx].compatibleUbuntuDistros,
+		);
+		if (compatibleUbuntuList.indexOf(ubuntuCodename) <= -1) {
+			throw new Error("Incompatible Gazebo and Ubuntu combination.");
```
Comment (j-rivero) [inline src/utils.ts:157]: Can we include in the error the webpage to look for the valid combinations? Probably https://gazebosim.org/docs/latest/getstarted/#step-1-install ?

File: src/utils.ts (hunk 2 above)
Comment (sauk2) [reply src/utils.ts]: Sounds like a great idea. I will create a separate issue for this

**Comment by sauk2** [comment]
> that uses directly the strings?

Thanks for the feedback! I have changed the object only to hold strings and removed the enums.

**Pull request merged**